scripts/download_poj.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Run script from code2seq dir using command:
+#    sh scripts/download_data.sh
+TRAIN_SPLIT_PART=$1
+VAL_SPLIT_PART=$2
+TEST_SPLIT_PART=$3
+DEV=$4
+SHUFFLE=$5
+DATA_DIR=./data
+DATASET_NAME=poj_104
+ASTMINER_PATH=../astminer/build/shadow/lib-0.5.jar
+SPLIT_SCRIPT=./scripts/split_dataset.sh
+
+DATA_PATH=${DATA_DIR}/${DATASET_NAME}
+
+if [ ! -d $DATA_DIR ]
+then
+  mkdir $DATA_DIR
+fi
+
+echo "Downloading dataset ${DATASET_NAME}"
+if [ -d "$DATA_PATH" ]
+then
+  echo "$DATA_PATH exists."
+else
+  if [ ! -f "$DATA_DIR/poj-104-original.tar.gz" ]
+  then
+    wget https://s3-eu-west-1.amazonaws.com/datasets.ml.labs.aws.intellij.net/poj-104/poj-104-original.tar.gz -P $DATA_DIR/
+  fi
+
+  echo "Unzip dataset"
+  tar -xvzf "$DATA_DIR/poj-104-original.tar.gz" -C $DATA_DIR/
+  mv "$DATA_DIR"/ProgramData "$DATA_PATH"
+
+  # In the developer mode we leave only several classes
+  if [ $DEV ]
+  then
+    echo "Dev mode"
+    find "$DATA_PATH"/* -type d -name "1[0-9][0-9]" -exec rm -rf {} \;
+    find "$DATA_PATH"/* -type d -name "[1-9][0-9]" -exec rm -rf {} \;
+  fi
```

Comment:
For tar, you can extract specific files too. See [this](https://unix.stackexchange.com/questions/35311/how-do-you-extract-a-single-folder-from-a-large-tar-gz-archive), for example.